pkg/cli/validate.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/client-lib/go/client"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/spf13/cobra"
+)
+
+// validateCmdOptions holds all the inputs and flags for the validate subcommand.
+type validateCmdOptions struct {
+	config *config.CLIConfig
+
+	// token flag to the command.
+	token string
+}
+
+// newValidateCmd creates a new subcommand for validating tokens.
+func newValidateCmd(cfg *config.CLIConfig) *cobra.Command {
+	opts := &validateCmdOptions{
+		config: cfg,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Validate the input token",
+		Long: strings.Trim(`
+Validate the given justification token and output the justifications and other
+standard claims if it's valid. If the token is invalid, an error will be returned.
+
+For example:
+
+    # Validate the justification token string
+    jvsctl validate --token "example token string"
+
+    # Validate the justification token read from pipe
+    cat token.txt | jvsctl validate --token -
+
+    # Output
+    ------BREAKGLASS------
+    false
+
+    ----JUSTIFICATION----
+    explanation  "test"
+
+    ---STANDARD CLAIMS---
+    aud  ["dev.abcxyz.jvs"]
+    iat  "2022-01-01T00:00:00Z"
+    iss  "jvsctl"
+    jti  "test-jwt"
+    nbf  "2022-01-01T00:00:00Z"
+    sub  "jvsctl"
+`, "\n"),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runValidateCmd(cmd, opts, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.token, "token", "t", "", `
+The JVS token that needs validation, can be passed as string or via pipe
+`)
+	cmd.MarkFlagRequired("token") //nolint // not expect err
+	// TODO: #140 add output format flag.
+	return cmd
+}
+
+func runValidateCmd(cmd *cobra.Command, opts *validateCmdOptions, args []string) error {
+	ctx := context.Background()
+
+	jvsclient, err := client.NewJVSClient(ctx, &client.JVSConfig{
+		Version:         "1",
+		JWKSEndpoint:    opts.config.JWKSEndpoint,
+		CacheTimeout:    5 * time.Minute,
```

Comment:
Two things here:

1. This should be a constant.
2. Does caching actually make sense? This isn't a long-running service, so the cache is effectively expired once the CLI exits since it's all in-memory.

Reply:
discussed offline, jvs config requires a >0 cache timeout, made it a constant and added comment.